versions.gradle

```diff
@@ -10,178 +10,17 @@ androidVersions.buildTools = '30.0.2'
-
-def androidX = [:]
-androidX.appcompat = "androidx.appcompat:appcompat:$versions.androidXAppCompat"
-androidX.browser = "androidx.browser:browser:$versions.androidXBrowser"
-androidX.cardview = "androidx.cardview:cardview:$versions.androidXCardView"
-androidX.constraint_layout = "androidx.constraintlayout:constraintlayout:$versions.constraint_layout"
-androidX.core = "androidx.core:core-ktx:$versions.androidXCore"
-androidX.dataStore = "androidx.datastore:datastore-preferences:$versions.androidXDataStore"
-androidX.design = "com.google.android.material:material:$versions.androidXMaterial"
-androidX.exifinterface = "androidx.exifinterface:exifinterface:$versions.androidXExifinterface"
-androidX.lifecycle = androidXLifecycle
-androidX.navigation = androidXNavigation
-androidX.percentlayout = "androidx.percentlayout:percentlayout:$versions.androidXPercentLayout"
-androidX.preference = "androidx.preference:preference:$versions.androidXPreferences"
-androidX.recyclerview = "androidx.recyclerview:recyclerview:$versions.recyclerView"
-androidX.room = androidXRoom
-androidX.test = androidXTest
-androidX.transition = "androidx.transition:transition:$versions.androidXTransition"
-androidX.viewPager2 = "androidx.viewpager2:viewpager2:$versions.androidXViewPager2"
-androidX.workManager = "androidx.work:work-runtime-ktx:$versions.archWorkManager"
-deps.androidX = androidX
-
-
-def playServices = [:]
-playServices.location = "com.google.android.gms:play-services-location:$versions.playServicesLocation"
-playServices.vision = "com.google.android.gms:play-services-vision:$versions.playServicesVision"
-deps.playServices = playServices
-
-def firebase = [:]
-firebase.analytics = "com.google.firebase:firebase-analytics:$versions.firebase_analytics"
-firebase.config = "com.google.firebase:firebase-config-ktx:$versions.firebase_config"
-firebase.crashlytics = "com.google.firebase:firebase-crashlytics:$versions.firebase_crashlytics"
-firebase.messaging = "com.google.firebase:firebase-messaging:$versions.firebase_messaging"
-deps.firebase = firebase
-
-def okhttp3 = [:]
-okhttp3.core = "com.squareup.okhttp3:okhttp:$versions.okhttp3"
-okhttp3.loggingIntercepror = "com.squareup.okhttp3:logging-interceptor:$versions.okhttp3"
-okhttp3.ok2CurlInterceptor = "com.github.mrmike:ok2curl:$versions.ok2CurlInterceptor"
-okhttp3.urlconnection = "com.squareup.okhttp3:okhttp-urlconnection:$versions.okhttp3"
-okhttp3.stetho = "com.github.mathroule.stetho:stetho-okhttp3:$versions.stetho"
-deps.okhttp3 = okhttp3
-
-def retrofit2 = [:]
-retrofit2.core = "com.squareup.retrofit2:retrofit:$versions.retrofit2"
-retrofit2.gsonAdapter = "com.squareup.retrofit2:converter-gson:$versions.retrofit2"
-deps.retrofit2 = retrofit2
-
-
-ext.deps = deps
-
 ext.buildscriptDeps = [:]
 buildscriptDeps.dependencyUpdates = "com.github.ben-manes:gradle-versions-plugin:$versions.dependencyUpdates"
 buildscriptDeps.detekt = "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:$versions.detekt"
```

Comment:
this can be declared in the catalog file, and then you can consume in the main build.gradle file